traits/ctraits.c

```diff
@@ -4243,16 +4245,41 @@ _trait_comparison_mode(trait_object *trait, PyObject *args)
         case 2:
             break;
         default:
-            return PyErr_Format(
+            PyErr_Format(
```

Comment:
The second `%d` should be replaced with `%ld` here, since `comparison_mode` is now a `long`.